File: benchmarks/java/integer_boolean/Java_Primitive_Integer/Implies.java
import de.hhu.stups.btypes.BInteger;
import de.hhu.stups.btypes.BBoolean;
import de.hhu.stups.btypes.BUtils;

public class Implies {






    private BInteger counter;

    public Implies() {
        counter = (BInteger) new BInteger(0);
    }

    public void simulate() {
        while((new BInteger(1).equal(new BInteger(1)).implies(counter.less(new BInteger(5000000)))).booleanValue()) {
            counter = (BInteger) counter.plus(new BInteger(1));
        }
    }

    public static void main(String[] args) {
        Implies exec = new Implies();
        long start = System.nanoTime();
        exec.simulate();
        long end = System.nanoTime();
        System.out.println(exec.getClass().toString() + " Execution: " + (end - start));
    }

}

File: benchmarks/cpp/integer_boolean/primitive_integer/Implies.cpp
#include <iostream>
#include <string>
#include "BUtils.cpp"
#include "BInteger.cpp"
#include "BBoolean.cpp"

#ifndef Implies_H
#define Implies_H

using namespace std;

class Implies {



    private:



        BInteger counter;

    public:

        Implies() {
            counter = (BInteger(0));
        }

        void simulate() {
            while(((BInteger(1)).equal((BInteger(1))).implies(counter.less((BInteger(5000000))))).booleanValue()) {
                counter = counter.plus((BInteger(1)));
            }
        }

};
int main() {
    clock_t start,finish;
    double time;
    Implies exec;
    start = clock();
    exec.simulate();
    finish = clock();
    time = (double(finish)-double(start))/CLOCKS_PER_SEC;
    printf("%f\n", time);
    return 0;
}
#endif

